### ISSUE TITLE
daily exx test failed

### ISSUE THREAD

**Issue by pxlxingliang** [deepmodeling/abacus-develop#4174]
### Describe the bug

The daily exx test at 20240516 is failed. Two of the examples raise errors related to memory.

385_NO_GO_RE_S1_HSE
```
 STEP OF RELAXATION : 1
 -------------------------------------------
 START CHARGE      : atomic
 DONE(12.5731    SEC) : INIT SCF
 ITER   ETOT(eV)       EDIFF(eV)      DRHO       TIME(s)    

malloc(): unaligned tcache chunk detected
```

386_NO_GO_MD_S1_HSE
```
 STEP OF MOLECULAR DYNAMICS : 0
 -------------------------------------------
 START CHARGE      : atomic
 DONE(11.6326    SEC) : INIT SCF
 ITER   ETOT(eV)       EDIFF(eV)      DRHO       TIME(s)    
 GE1    -4.655678e+02  0.000000e+00   2.249e-01  3.427e-01  
 GE2    -4.661634e+02  -5.956334e-01  1.188e-01  3.436e-01  
 GE3    -4.660560e+02  1.073815e-01   8.665e-02  3.480e-01  
...
 Updating EXX and rerun SCF	3.858e-03 (s)
 GE1    -4.659976e+02  0.000000e+00   1.132e-09  9.466e-01  

free(): unaligned chunk detected in tcache 2
```

https://app.bohrium.dp.tech/abacustest/?request=GET%3A%2Fapplications%2Fabacustest%2Fjobs%2Fsched-abacustest-exx-test-ea6c1c

[exx.zip](https://github.com/deepmodeling/abacus-develop/files/15328267/exx.zip)

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by pxlxingliang** [comment]
The test at 20240516 with commit c4db904 is passed. 
One of the commits after c4db904 may introduce some bugs:
24a27b96357ee101b17dec40f2c16d4e4ba69fac
cc96be0f255a1bbc7cf4a85c12ac0695dad2f8b9
b5ac150ba187cfb2c9d15c7acabdc46694eff5b8
a8e0d661f7aeaf5d5e0d6943be2862c4966ab41a
6138a50b95c195877c3d666288b0798ec699f0ea

**Comment by pxlxingliang** [comment]
I have run one of the failed example (385_NO_GO_RE_S1_HSE) on previous commits repeating 3 times, and only on commit https://github.com/deepmodeling/abacus-develop/commit/24a27b96357ee101b17dec40f2c16d4e4ba69fac, the example failed two times in the SCF process.
The two error messages are:
```
malloc_consolidate(): unaligned fastbin chunk detected
```
```
double free or corruption (fasttop)
malloc_consolidate(): unaligned fastbin chunk detected
```
The outputs: 
[exx (2).zip](https://github.com/deepmodeling/abacus-develop/files/15374033/exx.2.zip)



The timeline of these commits are:
![image](https://github.com/deepmodeling/abacus-develop/assets/91927439/6f8c8f76-2739-4b16-9a15-ca38d31d6f27)



We should check the commit https://github.com/deepmodeling/abacus-develop/commit/24a27b96357ee101b17dec40f2c16d4e4ba69fac again, which may be the origin of this issue.

**Comment by WHUweiqingzhou** [comment]
This issue has been solved by #4189
